desktop/core/src/desktop/js/apps/storageBrowser/FileChooserModal/FileChooserModal.tsx

```diff
@@ -0,0 +1,191 @@
+// Licensed to Cloudera, Inc. under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  Cloudera, Inc. licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+import React, { useCallback, useEffect, useMemo, useState } from 'react';
+import Modal from 'cuix/dist/components/Modal';
+import { Input, Spin, Tooltip } from 'antd';
+import Table, { ColumnProps } from 'cuix/dist/components/Table';
+import classNames from 'classnames';
+
+import FolderIcon from '@cloudera/cuix-core/icons/react/ProjectIcon';
+import { FileOutlined } from '@ant-design/icons';
+
+import { i18nReact } from '../../../utils/i18nReact';
+import useDebounce from '../../../utils/useDebounce';
+import useLoadData from '../../../utils/hooks/useLoadData';
+
+import { PathAndFileData } from '../../../reactComponents/FileChooser/types';
+import { VIEWFILES_API_URl } from '../../../reactComponents/FileChooser/api';
+import PathBrowser from '../../../reactComponents/FileChooser/PathBrowser/PathBrowser';
+
+import './FileChooserModal.scss';
+
+interface FileChooserModalProps {
+  onClose: () => void;
+  onSubmit: (destination_path: string) => void;
+  showModal: boolean;
+  title: string;
+  sourcePath: string;
+  submitText?: string;
+  cancelText?: string;
+}
+
+interface FileChooserTableData {
+  name: string;
+  path: string;
+  type: string;
+}
+
+const FileChooserModal = ({
```

Comment:
Like discussed on slack. The name is misleading if the only purpose is to let the user choose a destination path.

Reply:
@nidhibhatg Why not use the existing FileChooserModal that's under reactComponents/FileChooser? Could be extended to only show folders and limited to one FS.